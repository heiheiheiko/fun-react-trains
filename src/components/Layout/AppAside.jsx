import React from 'react';
import TrainFilterForm from './TrainFilterForm';
import MapZoomControls from './MapZoomControls';
import MapCenterLabel from './MapCenterLabel';

function AppAside(props) {
  return (
    <div className="overlay__layer overlay__layer--top-right mt-small">
      <div className="py-small pr-large pl-normal has-background-light">
        <TrainFilterForm/>
      </div>

      <div className="mt-small py-small pr-large pl-normal has-background-light">
        <MapZoomControls/>
      </div>

      <div className="mt-small py-small pr-large pl-normal has-background-light">
        <MapCenterLabel/>
      </div>
    </div>
   );
}

export default AppAside;
