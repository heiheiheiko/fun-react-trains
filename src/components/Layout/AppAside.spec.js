import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer'
import SpecContainer from '../../../__specs__/helpers/SpecIntelContainer';

import AppAside from './AppAside';

jest.mock('./TrainFilterForm', () => ()=> "TrainFilterForm");
jest.mock('./MapZoomControls', () => ()=> "MapZoomControls");
jest.mock('./MapCenterLabel', () => ()=> "MapCenterLabel");

describe('<AppAside>', () => {
  it('could be renderd', () => {
    const mountedSpecContainer = mount(<SpecContainer><AppAside/></SpecContainer>);
    expect(mountedSpecContainer.find(AppAside).length).toEqual(1);
  });

  it('matched the snapshot', () => {
    const specContainer = <SpecContainer><AppAside/></SpecContainer>;
    const renderedValue =  renderer.create(specContainer).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });
});