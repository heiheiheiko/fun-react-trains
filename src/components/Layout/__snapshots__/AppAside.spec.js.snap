// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppAside> matched the snapshot 1`] = `
<div>
   
  <div
    className="overlay__layer overlay__layer--top-right mt-small"
  >
    <div
      className="py-small pr-large pl-normal has-background-light"
    >
      TrainFilterForm
    </div>
    <div
      className="mt-small py-small pr-large pl-normal has-background-light"
    >
      MapZoomControls
    </div>
    <div
      className="mt-small py-small pr-large pl-normal has-background-light"
    >
      MapCenterLabel
    </div>
  </div>
   
</div>
`;
